webui/react/src/components/NavigationSideBar.tsx

```diff
@@ -255,6 +255,11 @@ const NavigationSideBar: React.FC = () => {
 
   if (!showNavigation) return null;
 
+  const openWorkspaceCreateModal = () => {
+    clusterStore.fetchKubernetesResourceManagers();
+    WorkspaceCreateModal.open();
+  };
```

Comment:
@ashtonG or @keita-determined (or anyone) should correct me if I'm wrong, but I think this should use [`useCallback`](https://react.dev/reference/react/useCallback) so that we don't get a brand new function value every time this component is rendered. Something like
```TypeScript
const openWorkspaceCreateModal = useCallback(() => {
  clusterStore.fetchKubernetesResourceManagers();
  WorkspaceCreateModal.open();
}, [clusterStore]); // not sure if WorkspaceCreateModal should be one of the dependencies
```

Reply:
don't think the stable reference provided by `useCallback` is necessary here